flink-datastream/src/main/java/org/apache/flink/datastream/impl/operators/ProcessOperator.java

```diff
@@ -46,6 +46,12 @@ public ProcessOperator(OneInputStreamProcessFunction<IN, OUT> userFunction) {
         chainingStrategy = ChainingStrategy.ALWAYS;
     }
 
+    @Override
+    public boolean isAsyncStateProcessingEnabled() {
+        // For normal operator (without keyed context) the async state processing is unused.
```

Comment:
That is true. I removed those. Thanks!

Reply:
I was wondering if it would be better to just inherit async on `KeyedProcessOperator`/`KeyedTwoInputNonBroadcastProcessOperator`/`KeyedTwoInputBroadcastProcessOperator`/`KeyedTwoOutputProcessOperator`?

BTW, should we offer an `toAsync()` method in `KeyedPartitionStream`?
which allows users to set the execution modes of different operators in a fine-grained manner.

Reply:
I'd prefer make Datastream V2 directly use the new state v2. We will provide fallback of sync state access in new state framework. So no `toAsync()` method and `AsyncKeyedStream` should be introduced. This would be more clear to users.

Of course, for Datastream V1, we won't touch the original state v1 code path, thus we should introduce new stream if we want to integrate state v2 into DS V1.